qiime2/plugins.py

```diff
@@ -37,6 +37,15 @@ def _init_metadata_(self, ref, factory):
         self._init_data_refs[ref] = factory
         return ref
 
+    def _init_data_collection_(self, ref, collection_type, *records):
+        self._init_data_refs[ref] = ref
+        collection = sorted([i.ref for i in records])
+        t = str(collection).replace("'", "")
```

Comment:
Nice! How about this modification?

```suggestion
        t = ', '.join(sorted([i.ref for i in records])
```

You'll have to tweak how you template in the set brackets, though...